### PR TITLE
User details page

### LINKED ISSUE #25939: Not readable style
### Description

The new style is so compact, that the table is not readable.

I would say, that the ID column is not needed.

### Screenshots

image

image

### Gitea Version

1.20.0

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

Ubuntu 20.04.4

### Browser Version

Chrome

### PR DESCRIPTION
**Pull request by denyskon**
This PR implements a proposal to clean up the admin users table by moving some information out to a separate user details page (which also displays some additional information).

Other changes:
  - move edit user page from `/admin/users/{id}` to `/admin/users/{id}/edit` -> `/admin/users/{id}` now shows the user details page
  - show if user is instance administrator as a label instead of a separate column
  - separate explore users template into a page- and a shared one, to make it possible to use it on the user details page
  - fix issue where there was no margin between alert message and following content on admin pages

<details>

<summary>Screenshots</summary>

grafik

grafik

</details>

Partially resolves #25939

### CODE REVIEW

**Comment by denyskon** [comment]
Btw, feel free to suggest other things that should be showed on this page :)

File: web_src/css/admin.css [hunk 1]
```diff
@@ -42,3 +42,15 @@
...
+.admin-responsive-columns {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1rem;
+  margin-bottom: 1rem;
+}
+
+.admin-responsive-columns > .column {
+  flex-grow: 1;
+  flex-basis: calc(50% - .5rem);
```
Comment (silverwind) [inline web_src/css/admin.css]: `flex: 1` should all that's necessary for equally sized boxes. If you need a gap, you should use `gap`.

File: templates/admin/user/view.tmpl [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+					<a class="ui primary tiny button" href="{{.Link}}/edit">{{ctx.Locale.Tr "admin.users.edit"}}</a>
+				</div>
+			</h4>
+			<div class="ui attached segment">
+				<div class="flex-list">
+					<div class="flex-item">
+						<div class="flex-item-leading">
+							{{ctx.AvatarUtils.Avatar .User 48}}
+						</div>
+						<div class="flex-item-main">
```
Comment (puni9869) [inline templates/admin/user/view.tmpl]: Is it possiable if we can divide this template into multiple `if possible`. `View` should be as modularize as possible.

File: web_src/css/admin.css (hunk 1 above)
Comment (denyskon) [reply web_src/css/admin.css]: Seems that you're right...... changed to `flex: 1`. I started with flex-basis 50%, then added a gap, then tried to make it responsible and in the end I had this :laughing:

File: templates/admin/user/view.tmpl (hunk 2 above)
Comment (denyskon) [reply templates/admin/user/view.tmpl]: I moved the user details block and the email list out to separate templates. Now, the view template basically only contains the layout, and the content is all being imported from subtemplates.

**Review by delvh — APPROVED** [review]
Code looks fine.
Only one small nit regarding the display:
The user link looks like it would redirect to the user profile, not to an info page about them.
Perhaps we should add an `information` icoin or something like that behind it to signify that you **should** click on the link, that it isn't just "decorative".
But that's not too urgent, we can still add that afterwards.

File: templates/admin/user/view_details.tmpl [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+				<b>{{ctx.Locale.Tr "admin.users.auth_source"}}:</b>
+				{{if eq .LoginSource.ID 0}}
+					{{ctx.Locale.Tr "admin.users.local"}}
+				{{else}}
+					{{.LoginSource.Name}}
+				{{end}}
+			</div>
+			<div class="flex-item-body">
+				<b>{{ctx.Locale.Tr "admin.users.activated"}}:</b>
+				<strong>
```
Comment (wxiaoguang) [inline templates/admin/user/view_details.tmpl]: `<strong>` with `svg`? It should be a no-op

File: templates/admin/user/view_details.tmpl [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			{{if .User.Location}}
+				<div class="flex-item-body">
+					<span class="flex-text-inline">{{svg "octicon-location"}}{{.User.Location}}</span>
+				</div>
+			{{end}}
+			{{if .User.Website}}
+				<div class="flex-item-body">
+					<span class="flex-text-inline">
+						{{svg "octicon-link"}}
+						<a target="_blank" rel="noopener noreferrer me" href="{{.User.Website}}">{{.User.Website}}</a>
```
Comment (wxiaoguang) [inline templates/admin/user/view_details.tmpl]: > The rel="me" attribute is used in RelMeAuth and Web sign in specifications as a way to enable a person to identify themselves to a web service using their domain name or a particular URL.

It doesn't seem right to use `me`

And, `noopener noreferrer` seems both no-op. 

* noopener: it's the default value for `target=_blank`
* noreferrer: the page head alread declares it:  `<meta name="referrer" content="no-referrer">`

File: templates/explore/user_list.tmpl [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+					{{end}}
+				</div>
+				<div class="flex-item-body">
+					{{if .Location}}
+						<span class="flex-text-inline">{{svg "octicon-location"}}{{.Location}}</span>
+					{{end}}
+					{{if and $.ShowUserEmail .Email $.IsSigned (not .KeepEmailPrivate)}}
+						<span class="flex-text-inline">
+							{{svg "octicon-mail"}}
+							<a href="mailto:{{.Email}}" rel="nofollow">{{.Email}}</a>
```
Comment (wxiaoguang) [inline templates/explore/user_list.tmpl]: `nofollow` is used for search-engines, and I guess no system would follow a `mailto:` link. So it seems a no-op?

File: web_src/css/admin.css [hunk 6]
```diff
@@ -42,3 +42,14 @@
...
+.admin-responsive-columns {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1rem;
+  margin-bottom: 1rem;
+}
+
+.admin-responsive-columns > .column {
+  flex: 1;
+}
```
Comment (wxiaoguang) [inline web_src/css/admin.css]: Maybe using `gt-f1` is simple enough?

File: templates/admin/user/list.tmpl [hunk 7]
```diff
@@ -68,36 +68,33 @@
 						</th>
 						<th>{{.locale.Tr "email"}}</th>
 						<th>{{.locale.Tr "admin.users.activated"}}</th>
-						<th>{{.locale.Tr "admin.users.admin"}}</th>
 						<th>{{.locale.Tr "admin.users.restricted"}}</th>
-						<th>{{.locale.Tr "admin.users.2fa"}}</th>
```
Comment (wxiaoguang) [inline templates/admin/user/list.tmpl:73]: Me as a site admin who also does security work, I would really like to see whether their account has setup 2FA

Comment (denyskon) [reply templates/admin/user/list.tmpl:73]: Does it really have to be in the users table? If you say it's needed we can bring it back, it doesn't consume that much space....

File: templates/explore/user_list.tmpl [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+					{{template "shared/user/name" .}}
+					{{if .Visibility.IsPrivate}}
+						<span class="ui basic tiny label">{{ctx.Locale.Tr "repo.desc.private"}}</span>
+					{{end}}
+				</div>
+				<div class="flex-item-body">
+					{{if .Location}}
+						<span class="flex-text-inline">{{svg "octicon-location"}}{{.Location}}</span>
+					{{end}}
+					{{if and $.ShowUserEmail .Email $.IsSigned (not .KeepEmailPrivate)}}
```
Comment (wxiaoguang) [inline templates/explore/user_list.tmpl]: I can see this template is reused, but would it cause over-abstraction?

For example, when the users are listed on the admin page, I guess it should always show user's email?

File: templates/admin/user/list.tmpl (hunk 7 above)
Comment (wxiaoguang) [reply templates/admin/user/list.tmpl:73]: In my personal option, I prefer to have it in the users table if there is enough space.

While it is not a must, if most people prefer not.

**Comment by denyskon** [comment]
@wxiaoguang all done :)

@delvh I'll maybe do it in a follow-up PR later at some point. I'm really sick currently and barely managed to apply wxiaoguang's suggestions :/

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**